atlasdb-config/src/main/java/com/palantir/atlasdb/factory/Leaders.java

```diff
@@ -99,13 +112,46 @@ public static LeaderElectionService create(
                 config.randomWaitBeforeProposingLeadershipMs(),
                 config.leaderPingResponseWaitMs());
 
-        env.register(ourAcceptor);
-        env.register(ourLearner);
+        return ImmutableLocalPaxosServices.builder()
+                .ourAcceptor(ourAcceptor)
+                .ourLearner(ourLearner)
+                .leaderElectionService(leader)
+                .pingableLeader(leader)
+                .build();
+    }
 
-        env.register(leader);
-        env.register(new NotCurrentLeaderExceptionMapper());
+    public static PaxosProposer createPaxosProposer(
+            PaxosLearner ourLearner,
+            List<PaxosAcceptor> acceptors,
+            List<PaxosLearner> learners,
+            int quorumSize,
+            ExecutorService executor) {
+        return PaxosProposerImpl.newProposer(
+                    ourLearner,
+                    ImmutableList.copyOf(acceptors),
+                    ImmutableList.copyOf(learners),
+                    quorumSize,
+                    executor);
+    }
 
-        return leader;
+    public static List<PaxosAcceptor> createAcceptorList(
```

Comment:
`createAcceptorList` and `createLearnerList` can be parameterized to `createList`?